### ISSUE TITLE
[LSP] Completion for Nickel keywords

### ISSUE THREAD

**Issue by ebresafegaga** [tweag/nickel#1095]

**Comment by yannham** [comment]
Is this really useful? I think keywords are usually small. I don't think rust-analyzer e.g. provides completion for keywords. Are there well-established language providing completion for keywords?

**Comment by ebresafegaga** [comment]
Yeah thinking about this now, I'm not sure this is really useful. But I think some languages (e.g F#, C#) provide completion for keywords.

**Comment by aspiwack** [comment]
I remembered something. Ocaml doesn't complete keywords, traditionally (I don't use Ocaml a lot of late), and it's annoying for one specific reason: whenever I type `in` in Emacs (like in `let … in`), it autocompletes to something from the standard library that starts in `in` (like `in_channel`, though I think it's another function). This is such a weird quirk. But there are certainly cheaper solutions than auto-completion of all keywords :slightly_smiling_face: .

**Comment by matthew-healy** [comment]
> I don't think rust-analyzer e.g. provides completion for keywords.

It does!

![Screenshot from 2023-02-17 13-21-39](https://user-images.githubusercontent.com/13350302/219652536-5efc5542-b533-4ad0-b100-e638b54e4338.png)
